docs/components/date_pickers.md

```diff
@@ -0,0 +1,319 @@
+<!--docs:
+title: "Material <component>"
+layout: detail
+section: components
+excerpt: "Date pickers let users select a date or range of dates."
+iconId: 
+path: /catalog/date-pickers/
+-->
+
+# Date pickers
+
+[Date pickers](https://material.io/components/date-pickers) let users select a date or range of dates. 
+
+![Mobile date picker for November 2018 with "17" selected.](assets/date_pickers/pickers_hero.png)
+
+## Contents
+
+- [Using date pickers](#using-pickers)
+- [Making date pickers accessible](#making-pickers-accessible)
+- [Date pickers](#date-pickers)
+- [Date range pickers](#date-range-pickers)
+- [Theming date pickers](#theming-pickers)
+
+## Using date pickers
+
+Date pickers let users select a date or range of dates. They should be suitable for the context in which they appear.
+
+Date pickers can be embedded into:
+
+- Dialogs on mobile
+- Text field dropdowns on desktop
+
+## Making date pickers accessible 
+
+Flutter's APIs support accessibility setting for large fonts, screen readers, and sufficient contrast. For more information, go to Flutter's [accessibility](https://flutter.dev/docs/development/accessibility-and-localization/accessibility) and [internationalization](https://flutter.dev/docs/development/accessibility-and-localization/internationalization) pages.
+
+## Date pickers
+
+Mobile date pickers can be used to select dates in the near future or past, when it’s useful to see them in a calendar month format. They are displayed in a dialog.
+
+Common use cases include:
+
+- Making a restaurant reservation
+- Scheduling a meeting
+
+They aren’t ideal for selecting dates in the distant past or future that require more navigation, such as entering a birth date or expiration date.
+
+Mobile date pickers allow selection of a single date and year.
+
+### Date pickers example
+
+`showDatePicker`
+- [Class definition](https://api.flutter.dev/flutter/material/showDatePicker.html)
+- [GitHub souce](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/pickers/date_picker_dialog.dart)
+- [Dartpad demo](https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/date_pickers/regular&gh_ref=develop)
+
+The following example shows a date picker.
+
+![Date picker with date set to November 17, 2020](assets/date_pickers/date_picker_example.png)
+
+```dart
+final DateTime newDate = await showDatePicker(
+  context: context,
+  initialDate: DateTime(2020, 11, 17),
+  firstDate: DateTime(2017, 1),
+  lastDate: DateTime(2022, 7),
+  helpText: 'Select a date',
+);
+```
+
+### Anatomy and key properties
+
+The following diagram shows the elements of a date picker:
+
+![Date pickers anatomy diagram](assets/date_pickers/pickers_anatomy.png)
+
+1. Title
+2. Selected date
+3. Switch-to-keyboard input icon
+4. Year selection menu
+5. Month pagination
+6. Current date
+7. Selected date
```

Comment:
Nit: In a lot of our docs we don't use 1.2.3.4. we do 1.1.1.1 Markdown is able to read it as a list and easier for us to maintain